src/shell.js

```diff
@@ -282,6 +282,7 @@ if (ENVIRONMENT_IS_SHELL) {
   if ((typeof process == 'object' && typeof require === 'function') || typeof window == 'object' || typeof importScripts == 'function') throw new Error('not compiled for this environment (did you build to HTML and try to run it not on the web, or set ENVIRONMENT to something - like node - and run it someplace else - like on the web?)');
 #endif
 
+#if ENVIRONMENT_MAY_BE_SHELL
```

Comment:
This is already inside such an ifdef, from line 278 - I guess you are trying to save the case of `ASSERTIONS`? We don't focus much on code size there, though, so I'm not sure if more ifdefs are worth it.

Reply:
I do look at the generated code in debug mode enough that I find this block confusing.  

I would love to remove the `|| ASSERTIONS` from line 278 but that proved a lot more complicated.

Reply:
Ok, fair enough, the downside of one ifdef sounds smaller than that.